server/src/models/swing.rs

```diff
@@ -110,3 +278,332 @@ fn decode(model: &[u8]) -> (f64, f64) {
+            &[42.0, 42.0, 42.8, 42.0, 42.0],
+            0.0
+        ))
+    }
+
+    #[test]
+    fn test_can_fit_sequence_of_different_values_with_error_bound_five() {
+        assert!(fit_sequence_of_values_with_error_bound(
+            &[42.0, 42.0, 42.8, 42.0, 42.0],
+            5.0
+        ))
+    }
+
+    fn fit_sequence_of_values_with_error_bound(values: &[Value], error_bound: Value) -> bool {
+        let mut model_type = Swing::try_new(error_bound).unwrap();
+        let mut fit_all_values = true;
+        let mut timestamp = FIRST_TIMESTAMP;
+        for value in values {
+            fit_all_values &= model_type.fit_value(timestamp, *value);
+            timestamp += SAMPLING_INTERVAL;
+        }
+        return fit_all_values;
+    }
+
+    // Tests for min().
+    proptest! {
+    #[test]
+    fn test_min(
+        first_value in num::i32::ANY.prop_map(i32_to_value),
+        final_value in num::i32::ANY.prop_map(i32_to_value)
+    ) {
+        let (a, b) = calculate_a_and_b(FIRST_TIMESTAMP, first_value as f64, FINAL_TIMESTAMP, final_value as f64);
+        let model = [a.to_be_bytes(), b.to_be_bytes()].concat();
+        let min = min(1, FIRST_TIMESTAMP, FINAL_TIMESTAMP, SAMPLING_INTERVAL as i32, &model, &[]);
+        prop_assert_eq!(min, Value::min(first_value, final_value));
+    }
+    }
+
+    // Tests for max().
+    proptest! {
+    #[test]
+    fn test_max(
+        first_value in num::i32::ANY.prop_map(i32_to_value),
+        final_value in num::i32::ANY.prop_map(i32_to_value)
+    ) {
+        let (a, b) = calculate_a_and_b(FIRST_TIMESTAMP, first_value as f64, FINAL_TIMESTAMP, final_value as f64);
+        let model = [a.to_be_bytes(), b.to_be_bytes()].concat();
+        let max = max(1, FIRST_TIMESTAMP, FINAL_TIMESTAMP, SAMPLING_INTERVAL as i32, &model, &[]);
+        prop_assert_eq!(max, Value::max(first_value, final_value));
+    }
+    }
+
+    // Tests for sum().
+    proptest! {
+    #[test]
+    fn test_sum(
+        first_value in num::i32::ANY.prop_map(i32_to_value),
+        final_value in num::i32::ANY.prop_map(i32_to_value)
+    ) {
+	// A segment of length one is used to have a known sum.
+        let (a, b) = calculate_a_and_b(FIRST_TIMESTAMP, first_value as f64, FINAL_TIMESTAMP, final_value as f64);
+        let model = [a.to_be_bytes(), b.to_be_bytes()].concat();
+        let sum = sum(1, FIRST_TIMESTAMP, FIRST_TIMESTAMP, SAMPLING_INTERVAL as i32, &model, &[]);
+        prop_assert_eq!(sum, first_value);
+    }
+    }
+
+    // Tests for grid().
+    proptest! {
+    #[test]
+    fn test_grid(value in num::i32::ANY.prop_map(i32_to_value)) {
+	// The linear function represents a constant value to have a known value.
+        let (a, b) = calculate_a_and_b(FIRST_TIMESTAMP, value as f64, FINAL_TIMESTAMP, value as f64);
+        let model = [a.to_be_bytes(), b.to_be_bytes()].concat();
+	let length = (((FINAL_TIMESTAMP - FIRST_TIMESTAMP) / SAMPLING_INTERVAL) + 1) as usize;
```

Comment:
Possible formatting error.

Reply:
The lines over 100 characters have been reformatted.